prov/efa/src/rdm/rxr_op_entry.c

```diff
@@ -198,88 +199,147 @@ void rxr_rx_entry_release(struct rxr_op_entry *rx_entry)
 }
 
 /**
- * @brief try to fill the desc field of a tx_entry
+ * @brief try to fill the desc field of an op_entry by memory registration
  *
- * The desc field of tx_entry contains the memory descriptors of
+ * The desc field of op_entry contains the memory descriptors of
  * the user's data buffer.
  *
  * For EFA provider, a data buffer's memory descriptor is a pointer to an
  * efa_mr object, which contains the memory registration information
  * of the data buffer.
  *
- * EFA provider does not require user to provide a descriptor, when
+ * EFA provider does not require user to provide a descriptor when
  * user's data buffer is on host memory (Though user can register
  * its buffer, and provide its descriptor as an optimization).
  *
- * The EFA device requires send buffer to be registered.
+ * However, there are a few occations that EFA device and shm
+ * require memory to be register with them:
  *
- * For a user that did not provide descriptors for the buffer,
- * EFA provider need to bridge the gap. It has 2 solutions for
- * this issue:
+ * First, when EFA device is used to send data:
+ *
+ *   If a non-read based protocol (such as eager, meidum, longcts)
+ *   is used, the send buffer must be registered with EFA device.
+ *
+ *   If a read based protocol is used, both send buffer
+ *   and receive buffer must be registered with EFA device.
+ *
+ * Seconde, when shm is used:
+ *   If eager protocol is used, no registration is needed (because
+ *   shm does not require registration for local buffer)
+ *
+ *   If a read based protocol is used, the send buffer must
+ *   be registered with shm, because send buffer is used as
```

Comment:
so shm does not require local buffer to be registered (with shm) for read?

Reply:
Yes. the local buffer is not registered.